lightningd/chaintopology.c

```diff
@@ -1413,6 +1417,15 @@ void setup_topology(struct chain_topology *topo)
 			blockscan_start = blocknum_reduce(blockscan_start, topo->ld->config.rescan);
 	}
 
+	fixup = db_get_intvar(topo->ld->wallet->db, "fixup_block_scan", -1);
+	if (fixup == -1) {
+		/* Never done fixup: this is set to non-zero if we have blocks. */
+		topo->old_block_scan = wallet_blocks_minheight(topo->ld->wallet);
```

Comment:
Would `when_lightning_became_cool` from chainparams be safe to use here?  @grubles points out that over half the blocks scanned on mainnet have no lightning txs to worry about.